cpp/models/lct_secir/parameters_io.h

```diff
@@ -509,10 +401,27 @@ IOResult<void> set_initial_data_from_confirmed_cases_ageres(Model& model,
     bool min_offset_needed_avail = false;
     bool max_offset_needed_avail = false;
 
-    // Go through the entries of rki_data and check if the entry has the age_group Group and if the date is
-    // needed for calculation. Confirmed cases are scaled by scale_confirmed_cases.
+    // Go through the entries of rki_data and check if the entry is age resolved and is referring to
+    // the age_group Group in the case with age resolution. If the date is
+    // needed for calculation, another function to handle the entry is called. Confirmed cases are scaled by scale_confirmed_cases.
     for (auto&& entry : rki_data) {
-        if ((size_t)entry.age_group == Group) {
+        if constexpr (is_age_resolved_entry_type<EntryType>::value) {
+            if ((size_t)entry.age_group == Group) {
```

Comment:
The code in the else block of the "if constexpr" is a duplicate of the code in the non-constexpr if block here. Since the inner if has no else attached to it, you can negate its condition and use a early return (early continue, to be exact):
```
if constexpr (...) {  // condition unchanged
    if (not ...) {
        continue; // skip to the next iteration of the for loop
    }
}
// contents of the else block follow here
int offset =  ...
```
This code is equivalent, but duplicate free with less indents/conditionals to keep track of.

Reply:
Thank you for this suggestion! I was aware that this is not an optimal solution, but I didn't know any better!